### PR TITLE
Adding "X-Pack/Basic Authentication" section to elasticsearch connector documentation

### PR DESCRIPTION
**Pull request by llmuculusll**
Adding "X-Pack/Basic Authentication" section to elasticsearch connector documentation according to #4165 pull request.

### CODE REVIEW

**Comment by llmuculusll** [comment]
I submitted CLA to cla@prestosql.io

File: presto-docs/src/main/sphinx/connector/elasticsearch.rst [hunk 1]
```diff
@@ -293,3 +293,15 @@ Property Name                                    Description
 ``elasticsearch.aws.secret-key``                 AWS secret key to use to connect to the Elasticsearch domain.
 ================================================ ==================================================================
 
+X-Pack/Basic Authentication
```
Comment (martint) [inline presto-docs/src/main/sphinx/connector/elasticsearch.rst:296]: This should say just "password authentication". "X-Pack" is an Elasticsearch extension framework and, as such, it's an implementation detail of Elasticsearch.

File: presto-docs/src/main/sphinx/connector/elasticsearch.rst [hunk 2]
```diff
@@ -293,3 +293,15 @@ Property Name                                    Description
...
+---------------------------
+
+To enable basic authentication using X-Pack, the ``elasticsearch.security`` option needs to be set to ``PASSWORD``.
```
Comment (martint) [inline presto-docs/src/main/sphinx/connector/elasticsearch.rst:299]:
```suggestion
To enable password authentication, the ``elasticsearch.security`` option needs to be set to ``PASSWORD``.
```

File: presto-docs/src/main/sphinx/connector/elasticsearch.rst [hunk 3]
```diff
@@ -293,3 +293,15 @@ Property Name                                    Description
...
+Additionally the following options need to be configured appropriately:
+
+================================================ ==================================================================
+Property Name                                    Description
+================================================ ==================================================================
+``elasticsearch.auth.user``                      Elasticsearch username for basic authentication.
+``elasticsearch.auth.password``                  Elasticsearch password for basic authentication.
```
Comment (martint) [inline presto-docs/src/main/sphinx/connector/elasticsearch.rst:306]:
```suggestion
``elasticsearch.auth.user``                      Elasticsearch username for password authentication.
``elasticsearch.auth.password``                  Elasticsearch password for password authentication.
```

**Comment by martint** [comment]
@cla-bot check

**Comment by martint** [comment]
Elasticsearch documentation has already been updated.

**Pull request closed without merging**